### PR TITLE
task DEV-1

### PR DESCRIPTION
**Pull request by ZheltokDina**

### CODE REVIEW

File: FindMaxSubstring/LengthChecker.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+namespace FindMaxSubstring
+{
+  /// <summary>
+  /// This class finds maximum number of the same elements in the substring
+  /// </summary>
+  public class LengthChecker
+  {
+    private string userString;
+    public string UserString
```
Comment (IgorKheidorov) [inline FindMaxSubstring/LengthChecker.cs]: Code style violation

File: .gitignore.txt [hunk 2]
```diff
@@ -1,6 +0,0 @@
-bin/
```
Comment (AlexanderSoroka) [inline .gitignore.txt]: should not be included to that changeset as it doesn't relate to the task

File: FindMaxSubstring/FindMaxSubstring.csproj.user [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (AlexanderSoroka) [inline FindMaxSubstring/FindMaxSubstring.csproj.user]: User settings must not be under git

File: FindMaxSubstring/WorkingWithString.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        {
+          throw new Exception("Your string is empty.");
+        }
+      }
+    }
+
+    /// <summary>
+    /// The trivial constructor with ToUpper() method 
+    /// </summary>
+    /// <param name="sequenceOfSymbols">string entered by the user</param>
```
Comment (AlexanderSoroka) [inline FindMaxSubstring/WorkingWithString.cs]: invalid argument name in comment

File: FindMaxSubstring/WorkingWithString.cs [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    public WorkingWithString(string userString)
+    {
+      this.SequenceOfSymbols = userString.ToUpper();
```
Comment (AlexanderSoroka) [inline FindMaxSubstring/WorkingWithString.cs]: It violates task specification as by default 'a' and 'A' are different symbols.

File: FindMaxSubstring/WorkingWithString.cs [hunk 6]
```diff
@@ -0,0 +1,71 @@
+﻿using System;
+
+namespace FindIdenticalConsecutiveCharacters
+{
+  /// <summary>
+  /// This class finds the maximum number of identical consecutive characters in the string
+  /// </summary>
+  public class WorkingWithString
```
Comment (AlexanderSoroka) [inline FindMaxSubstring/WorkingWithString.cs]: bad class name

File: FindMaxSubstring/WorkingWithString.cs [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        }
+      }
+      return maxLengthCount;
+    }
+
+    /// <summary>
+    /// Is used for console output 
+    /// </summary>
+    /// <param name="resultLenght">the resulting  number of identical consecutive characters</param>
+    public void OutputResult(int resultLenght)
```
Comment (AlexanderSoroka) [inline FindMaxSubstring/WorkingWithString.cs]: I doubt that this class should also take care about output

File: FindMaxSubstring/MaximumLengthChecker.cs [hunk 8]
```diff
@@ -0,0 +1,62 @@
+﻿using System;
+
+namespace FindIdenticalConsecutiveCharacters
+{
+  /// <summary>
+  /// This class finds the maximum number of identical consecutive characters in the string
+  /// </summary>
```
Comment (IgorKheidorov) [inline FindMaxSubstring/MaximumLengthChecker.cs]: Not checker, class find or count, but not check

File: FindMaxSubstring/MaximumLengthChecker.cs [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+    /// <summary>
+    /// This method is used for finding maximum length of identical consecutive characters in the string
+    /// </summary>
+    /// <returns>return maximum length of identical consecutive characters in the string</returns>
+    public int FindMaxLengthOfIdenticalConsecutiveSymbols()
+    {
+      int maxLengthCount = 0;
+      int currentCount = 1;
+      for (int i = 0; i < SequenceOfSymbols.Length - 1; i++)
```
Comment (IgorKheidorov) [inline FindMaxSubstring/MaximumLengthChecker.cs]: Logic error, find, if sequence length is 1

**Pull request merged**